feed/feed-inbound/feed-inbound-1.0.0/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java

```diff
@@ -0,0 +1,201 @@
+    private String feedType;
+    private Feed feed = null;
+    private java.util.Date lastUpdated;
+    private RssInject rssInject;
+    private FeedRegistryHandler feedRegistryHandler;
+    private String pathName;
+    private String dateFormat;
+    private DateFormat format;
+    private Parser parser;
+    private ParserOptions opts;
+
+    public ConsumeFeed(RssInject rssInject, long scanInterval, String host, String feedType,
+                       FeedRegistryHandler feedRegistryHandler, String name, String dateFormat) {
+        this.host = host;
+        this.feedType = feedType;
+        this.scanInterval = scanInterval;
+        this.rssInject = rssInject;
+        this.feedRegistryHandler = feedRegistryHandler;
+        this.pathName = name;
+        this.dateFormat = dateFormat;
+        this.scanIntervalDefined = scanInterval;
+        format = new SimpleDateFormat(FeedEPConstant.RSS_FEED_DATE_FORMAT, Locale.ENGLISH);
+        parser = Abdera.getNewParser();
+        //set filter
+        opts = parser.getDefaultParserOptions();
+        ListParseFilter filter = new WhiteListParseFilter();
+        if (feedType.equalsIgnoreCase(FeedEPConstant.FEED_TYPE_RSS)) {
+            filter.add(FeedEPConstant.FEED_RSS);
+            filter.add(FeedEPConstant.FEED_CHANNEL);
+            filter.add(FeedEPConstant.FEED_ITEM);
+            filter.add(FeedEPConstant.FEED_TITLE);
+            filter.add(FeedEPConstant.FEED_GUID);
+            filter.add(FeedEPConstant.FEED_PUBDATE);
+            filter.add(FeedEPConstant.FEED_LINK);
+        } else if (feedType.equalsIgnoreCase(FeedEPConstant.FEED_TYPE_ATOM)) {
+            filter.add(Constants.FEED);
+            filter.add(Constants.ENTRY);
+            filter.add(Constants.TITLE);
+            filter.add(Constants.ID);
+            filter.add(Constants.UPDATED);
+            filter.add(Constants.LINK);
+            filter.add(Constants.AUTHOR);
+        }
+        opts.setParseFilter(filter);
+    }
+
+    //check time interval
+    public void execute() {
+        try {
+            long currentTime = (new Date()).getTime();
+            if (((lastRanTime + scanInterval) <= currentTime)) {
+                lastRanTime = currentTime;
+                if (log.isDebugEnabled()) {
+                    log.debug("lastRanTime " + lastRanTime);
+                }
+                consume();
+            } else if (log.isDebugEnabled()) {
+                log.debug("Skip cycle since concurrent rate is higher than the scan interval : Feed Inbound EP " + pathName);
+            } else if (scanInterval < 999999999) {
+                scanInterval = scanInterval + 1000;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("End : Feed Inbound EP : " + pathName);
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    //consume feeds
+    private void consume() throws IOException, ParseException {
```

Comment:
Please catch the exception in this method instead of throws and catch, and In consume() method calling place you have caught generic exception.